modules/internal/Bytes.chpl

```diff
@@ -0,0 +1,473 @@
+
+      :returns: the index of the first occurrence of `needle` within a
+                the object, or 0 if the `needle` is not in the object.
+     */
+    proc find(needle: bytes, region: range(?) = 1:byteIndex..) : byteIndex {
+
+    }
+
+    /*
+      :arg needle: the :record:`bytes` to search for
+      :arg region: an optional range defining the indices to search within,
+                   default is the whole object. Halts if the range is not
+                   within ``1..bytes.length``
+
+      :returns: the index of the first occurrence from the right of `needle`
+                within the object, or 0 if the `needle` is not in the object.
+     */
+    proc rfind(needle: bytes, region: range(?) = 1:byteIndex..) : byteIndex {
+
+    }
+
+    /*
+      :arg needle: the :record:`bytes` to search for
+      :arg region: an optional range defining the substring to search within,
+                   default is the whole object. Halts if the range is not
+                   within ``1..bytes.length``
+
+      :returns: the number of times `needle` occurs in the object
+     */
+    proc count(needle: bytes, region: range(?) = 1..) : int {
+
+    }
+
+    /*
+      :arg needle: the :record:`bytes` to search for
+      :arg replacement: the :record:`bytes` to replace `needle` with
+      :arg count: an optional integer specifying the number of replacements to
+                  make, values less than zero will replace all occurrences
+
+      :returns: a copy of the object where `replacement` replaces `needle` up
+                to `count` times
+     */
+    proc replace(needle: bytes, replacement: bytes, count: int = -1) : bytes {
+
+    }
+
+    /*
+      Splits the :record:`bytes` on `sep` yielding the subbytes between each
+      occurrence, up to `maxsplit` times.
+
+      :arg sep: The delimiter used to break the object into chunks.
+      :arg maxsplit: The number of times to split the object, negative values
+                     indicate no limit.
+      :arg ignoreEmpty: * When `true`  -- Empty :record:`bytes` will not be
+                                          yielded, and will not count towards
+                                          `maxsplit`
+                        * When `false` -- Empty :record:`bytes` will be yielded
+                                          when `sep` occurs multiple times in a
+                                          row.
+     */
+    iter split(sep: bytes, maxsplit: int = -1,
+               ignoreEmpty: bool = false): bytes {
+
+    }
+
+    /*
+      Works as above, but uses runs of whitespace as the delimiter.
+
+      :arg maxsplit: The number of times to split the bytes, negative values
+                     indicate no limit.
+     */
+    iter split(maxsplit: int = -1) : bytes {
+
+    }
+
+    /*
+      Returns a new :record:`bytes` object, which is the concatenation of all of
+      the :record:`bytes` objects passed in with the receiving bytes inserted
+      between them.
```

Comment:
"the receiving bytes" seems confusing wording. Maybe "with the contents of the method receiver inserted between them" would be better? Anyway I think an example would help a lot.